tests/helpers.py

```diff
@@ -99,3 +102,22 @@ def produce(data, topic=config()['topics']['workspace_events']):
     producer.produce(topic, json.dumps(data), callback=_delivery_report)
     producer.flush()
     print(f"Produced {data}")
+
+
+@contextlib.contextmanager
+def set_env(**environ: Dict[str, str]):
+    """
+    Temporarily set the process environment variables.
+
+    Example:
+        with set_env(VAR_NAME='value'):
+            # Use VAR_NAME here
+            pass
+    """
+    old_environ = dict(os.environ)
+    os.environ.update(environ)
+    try:
```

Comment:
whats the behavior of this `try yield finally` block

Reply:
For contextmanager functions, they return a generator. The yield triggers the execution of the context manager's nested block:

```py
with set_env(blah="blah"):
    pass  # this indented block is executed on `yield` inside `set_env`
```

Everything after the `yield` is executed when the block is finished. It is wrapped in a `try/finally` because we don't want an error in the indented block to prevent the rest of our `set_env` function from running (that is, we always want to clear and restore the environment at the end)